11_FFTBloom/app_resources/fft_convolve_ifft.hlsl

```diff
@@ -0,0 +1,204 @@
+// where `Z` is the actual 0th row and `N` is the Nyquist row (the one with index FFT_LENGTH / 2). Those are packed together
+// so they need to be unpacked properly after FFT like we did earlier.
+
+struct PreloadedSecondAxisAccessor : PreloadedAccessorBase<ELEMENTS_PER_THREAD, _NBL_HLSL_WORKGROUP_SIZE_, scalar_t>
+{
+	int32_t mirrorWrap(int32_t paddedCoordinate)
+	{
+		const int32_t negMask = paddedCoordinate >> 31u;
+		const int32_t d = ((paddedCoordinate ^ negMask) / pushConstants.dataElementCount) ^ negMask;
+		paddedCoordinate = paddedCoordinate - d * pushConstants.dataElementCount;
+		const int32_t flip = d & 0x1;
+		return (1 - flip) * paddedCoordinate + flip * (pushConstants.dataElementCount - 1 - paddedCoordinate); //lerping is a float op
+	}
+
+	void preload(uint32_t channel)
+	{
+		// Set up accessor to point at channel offsets
+		bothBuffersAccessor = DoubleLegacyBdaAccessor<complex_t<scalar_t> >::create(getColMajorChannelStartAddress(channel), getRowMajorChannelStartAddress(channel));
+		const uint32_t padding = uint32_t(FFT_LENGTH - pushConstants.dataElementCount) >> 1;
+
+		for (uint32_t elementIndex = 0; elementIndex < ELEMENTS_PER_THREAD; elementIndex++)
+		{
+			const uint32_t index = _NBL_HLSL_WORKGROUP_SIZE_ * elementIndex | workgroup::SubgroupContiguousIndex();
+			const int32_t paddedIndex = index - int32_t(padding);
+			const int32_t wrappedIndex = mirrorWrap(paddedIndex);
+			preloaded[elementIndex] = bothBuffersAccessor.get(colMajorOffset(wrappedIndex, glsl::gl_WorkGroupID().x));
+		}
+	}
+
+	// Each element on this row is Nabla-ordered. So the element at `x' = index, y' = gl_WorkGroupID().x` that we're operating on is actually the element at
+	// `x = F(index), y = bitreverse(gl_WorkGroupID().x)` (with the bitreversal done as an N-1 bit number, for `N = log2(FFT_LENGTH)` *of the first axist FFT*)
+	template<typename SharedmemAdaptor>
+	void convolve(uint32_t channel, SharedmemAdaptor sharedmemAdaptor)
+	{
+		// Remember first row holds Z + iN
+		if (!glsl::gl_WorkGroupID().x)
+		{
+			for (uint32_t localElementIndex = 0; localElementIndex < ELEMENTS_PER_THREAD; localElementIndex += 2)
+			{
+				complex_t<scalar_t> zero = preloaded[localElementIndex];
+				complex_t<scalar_t> nyquist = getDFTMirror<SharedmemAdaptor>(localElementIndex, sharedmemAdaptor);
+
+				workgroup::fft::unpack<scalar_t>(zero, nyquist);
+
+				// We now have zero and Nyquist frequencies at NFFT[index], so we must use `getDFTIndex(index)` to get the actual index into the DFT
+				const uint32_t globalElementIndex = _NBL_HLSL_WORKGROUP_SIZE_ * localElementIndex | workgroup::SubgroupContiguousIndex();
+				const uint32_t indexDFT = workgroup::fft::FFTIndexingUtils<ELEMENTS_PER_THREAD, _NBL_HLSL_WORKGROUP_SIZE_>::getDFTIndex(globalElementIndex);
+
+				float32_t2 uv = float32_t2(indexDFT / float32_t(FFT_LENGTH), float32_t(0)) + pushConstants.kernelHalfPixelSize;
```

Comment:
the `indexDFT` should be `float32_t(indexDFT)`, otherwise compiler can start spitting out warnings about implicit promotions, range loss, etc.